### PR TITLE
Focus useful-libraries on packages

### PR DESCRIPTION
**Pull request by kwalrath**
The further down the page it gets, the less certain I am that I'm doing the right thing... But here's a first attempt. Please give me feedback!

### CODE REVIEW

**Comment by kwalrath** [comment]
Build failed due to link to a page that is introduced in #1510.

File: src/_guides/libraries/useful-libraries.md [hunk 1]
```diff
@@ -24,141 +32,55 @@ in alphabetical order:
...
-  gives examples of many commonly used dart:* APIs.
-* [Unboxing Packages]({{site.news}}/search/label/Unboxing%20Packages)
-  posts, written by written by Natalie Weizenbaum and published on the
-  [Dart news site.]({{site.news}})
-  This page links to some of Natalie's posts, but she covers other packages
-  not mentioned here, such as stream_channel, vm_service_client, and json_rpc_2.
+### Command-line and server packages
 
-### Packages contributed by the community
+The [Pub site]({{site.pub}}) doesn't currently support
```
Comment (mit-mit) [inline src/_guides/libraries/useful-libraries.md]: The 'other' category should work for this. There is no search filter for it, but we do badge those.

File: src/_guides/libraries/useful-libraries.md [hunk 2]
```diff
@@ -24,141 +32,55 @@ in alphabetical order:
...
+limiting your search to packages that support these kinds of apps.
+You can, however, search for words that describe the functionality you need.
 
-* [Pub site]({{site.pub}})
+Command-line apps often use the following packages:
```
Comment (mit-mit) [inline src/_guides/libraries/useful-libraries.md]: A few more ideas:

* archive
* intl
* yaml
* completion

Comment (mit-mit) [inline src/_guides/libraries/useful-libraries.md]: Maybe put these in a table and write a few words about what each package does?

File: src/_guides/libraries/useful-libraries.md [hunk 3]
```diff
@@ -24,141 +32,55 @@ in alphabetical order:
...
+* [path]({{site.pub-pkg}}/path)
+* [usage]({{site.pub-pkg}}/usage)
 
-* [Dart API]({{site.dart_api}}/{{site.data.pkg-vers.SDK.channel}}) contains the generated docs for dart:* libraries.
-* [Pub site]({{site.pub}}) hosts the generated docs for published packages.
-* [Flutter API]({{site.flutter_api}}) contains the generated docs for Flutter
-  libraries.
+Server apps can choose from many packages, including the following:
 
+* [aqueduct]({{site.pub-pkg}}/aqueduct)
```
Comment (mit-mit) [inline src/_guides/libraries/useful-libraries.md]: A few more:


* logging
* crypto
* appengine

File: src/_guides/libraries/useful-libraries.md (hunk 1 above)
Comment (kwalrath) [reply src/_guides/libraries/useful-libraries.md]: I couldn't figure out how to find packages that support OTHER.

File: src/_guides/libraries/useful-libraries.md (hunk 2 above)
Comment (kwalrath) [reply src/_guides/libraries/useful-libraries.md]: I don't want to double-cover packages, so I might mention the ones that are already in a table above (intl, yaml) but not include them in this table. What do you think?

**Comment by kwalrath** [comment]
@mit-mit ptal

File: src/_guides/libraries/useful-libraries.md (hunk 2 above)
Comment (mit-mit) [reply src/_guides/libraries/useful-libraries.md]: sgtm

**Pull request merged**